anndata/readwrite/read.py

```diff
@@ -55,14 +56,14 @@ def read_excel(
     """
     # rely on pandas for reading an excel file
     from pandas import read_excel
-    df = read_excel(fspath(filename), sheet)
-    X = df.values[:, 1:].astype(float)
+    df = read_excel(fspath(filename), sheet, dtype=dtype)
+    X = df.values[:, 1:].astype(dtype)
```

Comment:
why do we need the dtype in both lines? it should not be necessary in the second line anymore, should it?

Reply:
Agreed. Will remove it.